### ISSUE TITLE
generated_main.dart should using alias import

### ISSUE THREAD

**Issue by ekasetiawans** [flutter/flutter#100421]
## Use case

Since flutter generate `generated_main.dart` inside `.dart_tool/flutter_build` for every plugins that implement dartPluginClass, there was possible issue when two or more different plugins has same dartPluginClass.

## Proposal

To avoid conflicts in `_PluginRegistrant.register` method, we can use alias imports e.g
```
import 'package:foo/foo.dart' as p1;
import 'package:bar/bar.dart' as p2;
...
```

then we call registerWith method like this

```
p1.SomePluginClass.registerWith();
p2.SomePluginClass.registerWith();
```

**Comment by christopherfujino** [comment]
@stuartmorgan WDYT about this proposal. it sounds reasonable to me.

**Comment by stuartmorgan** [comment]
Yes, that was just an oversight. We should import `as {{pluginName}}` in the template.

**Comment by stuartmorgan** [comment]
@ekasetiawans Have you hit this with actual plugins in the wild? If so, we should prioritize fixing this.

**Comment by ekasetiawans** [comment]
@stuartmorgan Not yet, but this has the potential to become an issue. You can fix it later.
